target/linux/ramips/dts/rt5350_dlink_dir-506l.dts

```diff
@@ -0,0 +1,159 @@
+/dts-v1/;
+
+#include "rt5350.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "dlink,dir-506l", "ralink,rt5350-soc";
+	model = "D-Link DIR-506L";
+
+	aliases {
+		led-boot = &led_status_green;
+		led-failsafe = &led_status_red;
+		led-running = &led_status_green;
+		led-upgrade = &led_status_red;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		usb {
+			label = "dir-506l:green:usb";
+			gpios = <&gpio0 9 GPIO_ACTIVE_LOW>;
+		};
+
+		led_status_green: status_green {
+			label = "dir-506l:green:status";
+			gpios = <&gpio0 19 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_status_red: status_red {
+			label = "dir-506l:red:status";
+			gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		internet {
+			label = "dir-506l:green:internet";
+			gpios = <&gpio1 0 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		low_battery {
+			label = "low_battery";
+			gpios = <&gpio0 10 GPIO_ACTIVE_HIGH>;
+			linux,code = <BTN_0>;
+		};
+
+		charging {
+			label = "charging";
+			gpios = <&gpio0 11 GPIO_ACTIVE_HIGH>;
+			linux,code = <BTN_1>;
+		};
+
+		on_battery {
+			label = "on_battery";
+			gpios = <&gpio0 12 GPIO_ACTIVE_HIGH>;
+			linux,code = <BTN_2>;
+		};
+	};
+
+	gpio_export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		usb_power {
+			gpio-export,name = "usb_power";
+			gpio-export,output = <1>;
+			gpios = <&gpio0 18 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "jboot";
+				reg = <0x0 0x10000>;
+				read-only;
+			};
+
+			partition@10000 {
+				compatible = "amit,jimage";
+				label = "firmware";
+				reg = <0x10000 0x7e0000>;
+			};
+
+			config: partition@7f0000 {
+				label = "config";
+				reg = <0x7f0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&state_default {
+	gpio {
+		groups = "i2c", "jtag", "uartf", "led";
+		function = "gpio";
+	};
+};
+
+&ethernet {
+	mtd-mac-address = <&config 0xe084>;
+	mtd-mac-address-increment = <(1)>;
```

Comment:
Are there other locations for the MAC address? It's particularly strange that all cases require and increment. Anything in the jboot  partition?

Reply:
This is correct MAC address, have no idea why increment required to make it euqual with MAC printed on label.

Reply:
That's why I suspect there will be other "correct" addresses somewhere.

Reply:
http://www.netlab.linkpc.net/download/hardware/DIR-506L/2.15.bin this is dump, I see only one MAC address, decremented by 1 from value printed on label:90:94:e4:ea:15:60.

Reply:
Thanks. It appears you are right, though I really don't understand the vendor's intentions here.